saltyrtc/server/server.py

```diff
@@ -231,6 +232,34 @@ def handler(self, connection, ws_path):
         # Remove client from path
         path.remove_client(client)
 
+        # Send disconnected message if client was authenticated
+        if client.authenticated:
+            # Initiator: Send to all responders
+            if client.type == AddressType.initiator:
+                responder_ids = path.get_responder_ids()
+                coroutines = []
+                for responder_id in responder_ids:
+                    responder = path.get_responder(responder_id)
+
+                    # Create message and add send coroutine to task queue of the responder
+                    message = DisconnectedMessage.create(
+                        AddressType.server, responder_id, client.id)
+                    responder.log.debug('Enqueueing disconnected message')
+                    coroutines.append(responder.enqueue_task(responder.send(message)))
+                yield from asyncio.gather(*coroutines)
```

Comment:
Is the `yield from` necessary here, or could we drop it? The code seems to work without it.

Reply:
Yes, `enqueue_task` returns a coroutine and simply enqueues the task. It can (theoretically) block when the queue is full (although it has no restriction at the moment).

Reply:
Don't forget to pass in `loop=...`.

Reply:
Yes to keeping or dropping the "yield from"? :slightly_smiling_face:

Reply:
Keep it.